nntrainer/layers/lstmcell.cpp

```diff
@@ -23,365 +25,315 @@ namespace nntrainer {
 
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 
-enum LSTMParams {
-  weight_xh,
+enum LSTMCellParams {
+  weight_ih,
   weight_hh,
-  bias_h,
+  bias_ih,
   hidden_state,
-  mem_cell,
-  fgio,
+  cell_state,
+  ifgo,
   dropout_mask
 };
 
+const std::vector<unsigned int>
+getInOutIdx(std::array<unsigned int, 7> &wt_idx) {
+  std::vector<unsigned int> ret(3);
+  ret[0] = SINGLE_INOUT_IDX;
+  ret[1] = wt_idx[LSTMCellParams::hidden_state];
+  ret[2] = wt_idx[LSTMCellParams::cell_state];
+  return ret;
+}
+
+const std::vector<unsigned int>
+getTensorIdx(std::array<unsigned int, 7> &wt_idx) {
+  std::vector<unsigned int> ret(1);
+  ret[0] = wt_idx[LSTMCellParams::ifgo];
+  return ret;
+}
+
 LSTMCellLayer::LSTMCellLayer() :
   LayerImpl(),
-  lstm_props(props::Unit(), props::HiddenStateActivation(),
-             props::RecurrentActivation(), props::DropOutRate(),
-             props::MaxTimestep(), props::Timestep()),
+  lstmcell_props(props::Unit(), props::DropOutRate(), props::MaxTimestep(),
```

Comment:
For later) any reason to delete setting hiddenStateActivation / RecurrentActivation?

Reply:
Self answered: This is delegated to lstm cell core